src/Backups/BackupIO_S3.cpp

```diff
@@ -239,6 +245,11 @@ void BackupWriterS3::copyObjectMultipartImpl(
 
         auto etag = outcome.GetResult().GetCopyPartResult().GetETag();
         part_tags.push_back(etag);
+
+        position = next_position;
+
+        if (part_number % rw_settings.upload_part_size_multiply_parts_count_threshold == 0)
+            upload_part_size *= rw_settings.upload_part_size_multiply_factor;
```

Comment:
We also must limit upload_part_size to 5GB at most.

Reply:
Maybe with default `upload_part_size_multiply_parts_count_threshold` it's impossible to exceed, but still.

Reply:
done